packages/mediapipe-core/test/task_options_test.dart

```diff
@@ -0,0 +1,116 @@
+import 'dart:ffi';
+import 'dart:typed_data';
+
+import 'package:test/test.dart';
+import 'package:mediapipe_core/mediapipe_core.dart';
+
+void main() {
+  group('BaseOptions constructor should', () {
+    test('enforce exactly one of modelPath and modelBuffer', () {
+      expect(
+        () => BaseOptions(
+          modelAssetPath: 'abc',
+          modelAssetBuffer: Uint8List.fromList([1, 2, 3]),
+        ),
+        throwsA(TypeMatcher<AssertionError>()),
+      );
+
+      expect(BaseOptions.new, throwsA(TypeMatcher<AssertionError>()));
+    });
+  });
+
+  group('BaseOptions.toStruct/fromStruct should', () {
+    test('allocate memory in C for a modelAssetPath', () {
+      final options = BaseOptions(modelAssetPath: 'abc');
+      final struct = options.toStruct();
+      expect(toDartString(struct.ref.model_asset_path), 'abc');
+
+      // TODO: Is this the most precise expectation we can make here?
+      expect(struct.ref.model_asset_buffer, isA<Pointer<Never>>());
```

Comment:
Good question! I have not seen an expect like this. What is it verifying?

Reply:
That the object is a null pointer, but sadly, more than just null pointers will pass this assertion for implementation detail reasons that the straightening out of which remain a backlog improvement item for `ffigen` in general. This is really a question for @dcharkes.

Reply:
Try `.address == 0`

`Never` is a bottom type in the type hierarchy if I'm not mistaken, so any type argument will pass. (Moreover, at some point we want to actually get rid of the runtime type argument of `Pointer`.)